shapely/geometry/base.py

```diff
@@ -154,75 +142,42 @@ class BaseGeometry(object):
     #     Cached ctypes pointer to GEOS geometry. Not to be accessed.
     # _geom : c_void_p
     #     Property by which the GEOS geometry is accessed.
-    # __p__ : object
-    #     Parent (Shapely) geometry
     # _ndim : int
     #     Number of dimensions (2 or 3, generally)
-    # _crs : object
-    #     Coordinate reference system. Available for Shapely extensions, but
-    #     not implemented here.
-    # _other_owned : bool
-    #     True if this object's GEOS geometry is owned by another as in the
-    #     case of a multipart geometry member.
-    __geom__ = EMPTY
-    __p__ = None
-    _ndim = None
-    _crs = None
-    _other_owned = False
-    _is_empty = True
 
     # Backend config
     impl = DefaultImplementation
 
-    # a reference to the so/dll proxy to preserve access during clean up
-    _lgeos = lgeos
+    def __new__(self):
+        # TODO create empty geometry - should we deprecate this constructor?
+        return pygeos.from_wkt("GEOMETRYCOLLECTION EMPTY")
 
-    def empty(self, val=EMPTY):
-        if not self._other_owned and self.__geom__ and self.__geom__ != EMPTY:
-            try:
-                self._lgeos.GEOSGeom_destroy(self.__geom__)
-            except (AttributeError, TypeError):
-                # _lgeos might be empty on shutdown
-                log.exception("Failed to delete GEOS geom")
+    @property
+    def _geom(self):
+        return self._ptr
 
-        self._is_empty = True
-        self.__geom__ = val
+    @property
+    def __geom__(self):
+        return self._ptr
```

Comment:
`__geom__` is another ["dunder"](https://docs.python.org/3/reference/lexical_analysis.html#reserved-classes-of-identifiers) I think we could remove for shapely-2.0, as it's not a system-defined name. I see that this holds the same value as `_geom` and `_ptr` (from pygeos).

Reply:
Yes, the attributes giving read access to the pointer (`_geom`, `__geom__`, and now also `_ptr`) can use a clean-up. But for now, those are kept here in any case for compatibility with the rest of shapely that is not yet adapted.

So we should probably decide which one to keep (eg `_geom`) and deprecate the other already in Shapely 1.8 (eg `__geom__`)

(might also be worth a separate issue, as it is not "solvable" for this PR)

Reply:
Opened https://github.com/Toblerity/Shapely/issues/999 for this `__geom__` / `_geom` dicussion